riscv/processor.cc

```diff
@@ -536,6 +539,11 @@ void state_t::reset(processor_t* const proc, reg_t max_isa)
   if (proc->extension_enabled(EXT_ZCMT))
     csrmap[CSR_JVT] = jvt = std::make_shared<jvt_csr_t>(proc, CSR_JVT, 0);
 
+  if (proc->extension_enabled(EXT_ZICFISS)) {
+    reg_t ssp_mask = -reg_t(xlen == 64 ? 8 : 4);
```

Comment:
One last change.  Now that you've made the code clearer, an even better improvement reveals itself: This is really `-reg_t(xlen / 8)`.